Add German Translations

ffried opened the pull request:
... and a `.gitignore`.

The German translation is probably not final as-is. There are many aspects where I either had no idea in what context they are used, or had no idea about what it actually means at all (and couldn't find anything - probably also due to lack of context).
So it would be good if you can find someone to look over it and correct any mistakes. But it should serve as a good starting point with many obvious texts being done. I'd expect the changes to only be within the nitty gritty details.

I'd be happy to look through it, if there's a test version with the German translation, though.


<hr/>

**Note:** In German as used in Germany and Austria, there is a the `ß` character, that is used in some places to mark a different pronunciation of `ss`. This is not used in Switzerland and Liechtenstein (which also use German), however. I initially planned to duplicate the translation for the former countries, adding `ß` where needed, but it's actually easier the other way round.

HarshilShah commented:
Hey Florian, thank you so much for this PR! I've asked on Twitter/Mastodon/etc. to see if anyone else can give it a look, too!

Please let me know what the ambiguities are, I'd love to add comments and improve the experience for translators!

As for units, Apple does provide some really good APIs for formatting `Measurement`s, but they are a bit limited in terms of how you can extend them for things like steps, workouts, etc., and you also can't extract the correctly pluralised unit text alone, which is useful for contexts like text fields, where only the number is editable, and the unit needs to be displayed separately, and so I've added manual localisation strings for those cases

HarshilShah commented:
Just adding an Xcloc export of this branch so folks reviewing it can open it without downloading Xcode: [de.xcloc.zip](https://github.com/HarshilShah/PeakLocalisations/files/13945633/de.xcloc.zip)

dhoepfl commented on PeakLocalisations/Localisation/App Store/InfoPlist.xcstrings:
```diff
@@ -32,6 +51,12 @@
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Peak schreibt dein Gewicht und deine Körpermasse in HealthKit"
```
Should be “Körpermaße”.

ffried replied:
Fixed.

ffried commented:
I've just revised a few translations that came to me over night (also fixed a few mistakes I spotted when looking through them again)...

Also, here's a few examples of where translating is difficult (including reasons):

- `Formatting`: A lot of duplicated texts, that are essentially all the same. Some of which I understand, some I think could be simplified (e.g. using the attributed variants in non-attributed places, by not adding any attributes?). Also, it might make sense to add a comment to the attributed variants, explaining how they are structured - might look cryptic to people not familiar with `AttributedString`...
- `AddEntry`: `Log %@` -> That's a tricky one. Your English translation says "Log Your %@". However, the translation of "Your" in German might vary depending on what is inserted for `%@`.
- `Details`: `AddBlock.Title` -> It would be helpful to know what is inserted for `%@` - even though it likely won't affect the German translation here. Same for `RemoveBlock.Title`.
-  `PeakPro`: `Reason.Themes` and `Reason.Widgets` -> These look like prefixes, but what coming after? (I assumed "Peak Pro").
- `Widget`: `%@ Hold to set up` and `%@ Tap and hold to set up`: A comment describing what is inserted for `%@` would help
- `Widget`: `None` -> The German translation could actually vary (e.g. between "Nichts" or "Keine") based on what "None" means here. In this case it seems to be related to "Metrics" (as in "No Metric"). Also note, that in case of "Keine", it is even more related to the context it is used in, since there are variations ("Keines", "Keiner", ...).
- `Goals`: `Stat.DaysInHand`: Not sure in what context this is used in.

I think that's most of what I struggled with.

Another thing that's tricky with German (but likely other languages as well), is that there might be multiple translations for one word - especially words that are nowadays also understood in English. One example is "Theme": you could translate it as "Stil", but "Theme" itself is also understood (and in this case probably more appropriate, which is why I just changed that).
This is not a bad thing per se, but given that the word appears in multiple translations (across multiple files) you need to keep in mind what choice you made, so that it's consistent across all files.
Many words are translated multiple times (and I mostly do understand why!), which makes this even harder.

<hr/>
<br/>
To sum it up, apart from the list above, I'm not sure how much can be done to simplify translations, though. I just wanted to share what I noticed during translation.

HarshilShah commented:
Thank you so much for these notes! I'll try to add some comments in there to clarify a lot of this!

About the formatting, when I try to instantiate the attributed string key with a regular string, it seems to decode the attributes as regular text, but let me see if that's something I can work around somehow.

As for the `Log` title, that will be weight, BMI, lean body mass, or body fat percentage, basically prefilling in the title from there. Do you think it needs separate strings? And for `Add Block` it'll be Chart, Goal, Benchmark, Trends, etc., so please let me know if that needs separate strings too

> * `Widget`: `%@ Hold to set up` and `%@ Tap and hold to set up`: A comment describing what is inserted for `%@` would help

The substitutions here are an SF Symbol for "hand.point.up.left"

<img width="353" alt="Screenshot 2024-01-16 at 2 16 34 PM" src="https://github.com/HarshilShah/PeakLocalisations/assets/3503595/4a0c623a-57cd-4b54-8017-b1712bffa853">

As for `Goals.Stat.DaysInHand`, yeah that's a tricky one for me to explain in English too, but basically it says how many days you have left in the period (week/month). This includes today if you haven't met the goal yet (and so it's still "in hand"), but if you have, today isn't counted.

ffried commented:
> As for the Log title, that will be weight, BMI, lean body mass, or body fat percentage, basically prefilling in the title from there. Do you think it needs separate strings? 

Then definitively yes. For example, with "Weight" it would be "Trage dein Gewicht ein", but "Log Your Lean Body Mass" would be "Trage deine magere Körpermasse ein". Notice that the "m" in "magere" is lowercase (but uppercase in the title alone), and also it's "deine" vs. "dein".
For the `Add Block`, I've adjusted the German translation so that it shouldn't matter. Though I can imagine other languages not having that luxury...

`Goals.Stat.DaysInHand` -> Ok, then I got that right. Btw. I think it should be "Days At Hand" in English.

The SF Symbol that is inserted is a very important comment. That explains why the text starts with a capital letter (something @dhoepfl also mentioned on Mastodon).

ffried commented:
To have it documented here as well: https://applelocalization.com is a great resource for checking how Apple translates something. Thanks @dhoepfl for [pointing this out on Mastodon](https://layer8.space/@teilweise/111764759683187961).

I've used it to align many translations with Apple's now. Especially for the workout types.

HarshilShah commented:
> Then definitively yes. For example, with "Weight" it would be "Trage dein Gewicht ein", but "Log Your Lean Body Mass" would be "Trage deine magere Körpermasse ein". Notice that the "m" in "magere" is lowercase (but uppercase in the title alone), and also it's "deine" vs. "dein".

Gotcha, I've just updated `main` to account for this! It's now 4 separate specialised strings instead of one

ffried commented:
> Gotcha, I've just updated main to account for this! It's now 4 separate specialised strings instead of one

Great, thanks. Just added the translations for these as well.
I think this looks good now from my side.

Manuel-Kehl commented on PeakLocalisations/Localisation/App Store/Metadata.xcstrings:
```diff
@@ -16,6 +22,12 @@
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Mit Peak kannst du deine eigene Fitness-Übersicht erstellen und mit einer Reihe von Widgets immer den Überblick behalten.\n\n→ Deine persönliche Fitness-Übersicht\nErstelle eine individuelle Übersicht, das auf deine Bedürfnisse zugeschnitten ist, indem du genau die Kennzahlen auswählst, die du möchtest, z. B. Schritte, Trainings, Aktivitätsringe und mehr. Jede Kennzahl kann ...
```
Since @HarshilShah was asking around for German speakers to do a quick double-check, I had a brief skim and only found one potential typo:

@ffried, I assume you meant to say "Erstelle eine individuelle Übersicht, ~das~ **die** auf deine Bedürfnisse zugeschnitten ist, ...", right?

Apart from this the translation looks great 👍

ffried replied:
Good catch, thanks! That's a leftover from changing "Dashboard" to "Übersicht"...
Fixed now.

HarshilShah commented:
I guess if everyone's happy with the translations right now, we're good to merge?

HarshilShah commented:
And merged! Thank you to everyone for helping with this!

ffried commented:
@arik-so thanks a lot for your comments! I'm AFK today, but I'll integrate them in the coming days!

ffried commented on PeakLocalisations/Localisation/App Store/Metadata.xcstrings:
```diff
@@ -16,6 +22,12 @@
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Mit Peak kannst du deine eigene Fitness-Übersicht erstellen und mit einer Reihe von Widgets immer den Überblick behalten.\n\n→ Deine persönliche Fitness-Übersicht\nErstelle eine individuelle Übersicht, die auf deine Bedürfnisse zugeschnitten ist, indem du genau die Kennzahlen auswählst, die du möchtest, z. B. Schritte, Trainings, Aktivitätsringe und mehr. Jede Kennzahl kann ...
```
I think the plural is correct here, since it is about multiple workouts, not just one. And even though German imported that word, the plural is still "Trainings".

As for the added `an`, it would change the meaning of the sentence. It's about Peak feeling "right at home" (original) on your iPhone, since it uses the same patterns. It's more in the sense of 'it fits right in'.

ffried commented on PeakLocalisations/Localisation/App Store/InfoPlist.xcstrings:
```diff
@@ -53,16 +84,45 @@
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Etwas stimmt nicht?"
```
Good point, changed.

ffried commented on PeakLocalisations/Localisation/App Store/InfoPlist.xcstrings:
```diff
@@ -32,6 +51,12 @@
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Peak schreibt dein Gewicht und deine Körpermaße in HealthKit"
```
Yea, I mass-changed "ss" to "ß" (see older review comments), and went over it too quickly apparently. Thanks for catching this!

ffried commented on PeakLocalisations/Localisation/App Store/Screenshots.xcstrings:
```diff
@@ -59,6 +89,12 @@
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Blicke auf deinen Fortschritt 2023 zurück"
```
I'm not a big fan of using only nouns. Using the verb "zurückblicken" sounds more natural IMHO. I added an `in`, because you're not looking at the progress of the year itself, but at the progress _you_ made _within_ that year.

ffried commented on PeakLocalisations/Localisation/App Store/Screenshots.xcstrings:
```diff
@@ -103,6 +157,12 @@
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "...Und für deinen Sperrbildschirm"
```
It is a continuation of the previous sentence that still somewhat stands on its own (it's used in screenshots and both texts are used in a separate screenshot where the second follows the first).

<img width="892" alt="image" src="https://github.com/HarshilShah/PeakLocalisations/assets/1689782/28ccd71e-6b04-40ab-b23b-bd8638c972a4">

But your points are valid. Btw. this is the only text that is not using the unicode character in English either, so that's why I kept the three dots.
@HarshilShah is there a specific reason for this here in English or just an oversight?

HarshilShah replied:
Yeah definitely just an oversight, will fix

ffried commented on PeakLocalisations/Localisation/Features/WidgetSetup.xcstrings:
```diff
@@ -62,6 +92,12 @@
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Drücke lange auf einen freien Bereich auf deinem Sperrbildschirm.\nTippe auf \"Anpassen\".\nWähle den Sperrbildschirm aus.\nWähle ein Rechteck über oder unter der Zeit aus.\nSuche \"Peak\".\nWähle irgendein Widget aus um es hinzuzufügen (Hinweis: Du musst eventuell existierende Widgets entfernen, wenn nicht genug Platz vorhanden ist)\nTippe auf das Widget, um es zu bearbeit ...
```
I changed it to "Wähle eines der Widgets aus" - sounds a bit more on point to me (since you're presented with a list of widgets at that point).

ffried commented on PeakLocalisations/Localisation/Features/WidgetSetup.xcstrings:
```diff
@@ -178,6 +268,12 @@
+        "de" : {
+          "stringUnit" : {
+            "state" : "translated",
+            "value" : "Hinweis: Der StandBy Modus funktioniert nur auf einem iPhone mit iOS 17 oder neuer.\nGehe in die Einstellungen > StandBy und stelle sicher, dass StandBy aktiviert ist.\nVerbinde dein iPhone mit einem Ladegerät und kippe es in eine horizontale Position.\nDrücke den Seitenschalter.\nStreiche links oder rechts, um zwischen den verschiedenen Ansichten zu wechseln. Streiche hoch ...
```
I actually changed it to just "StandBy" (as Apple does). Search for "StandBy" on the [official Apple iOS 17 page](https://www.apple.com/de/ios/ios-17/) and click on the plus in the lower right of the picture to see a text.

Similar for the reference to the minimum requirements btw. IMHO it's more natural to reference a single device than multiple ones, since most people only have one (active) iPhone.
